### PR TITLE
Card ID field changes in create task form

### PR DESCRIPTION
**Pull request by Angatupyry**
## What's new

- Make card ID field freesolo
- Options for card ID in resources

## Self-checks

- [ ] I have prototyped this new feature (if necessary) on Figma 
- [ ] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [ ] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

**Review by aaronchongth — CHANGES_REQUESTED** [review]
Thanks for handling this! Just have a few comments regarding naming, the rest are all good

File: packages/react-components/lib/tasks/create-task.tsx [hunk 1]
```diff
@@ -364,21 +366,40 @@ function DeliveryCustomTaskForm({
...
+            pickup_activity.description.cart_rfid = parseInt((ev.target as HTMLInputElement).value);
+            onInputChange({
+              ...taskDesc,
+              activities: [pickup_activity, taskDesc.activities[1]],
+            });
+          }}
+          renderInput={(params) => (
+            <TextField
+              {...params}
+              label="Cart RFID"
```
[the remark this answers is not in the document]
Comment (Angatupyry) [reply packages/react-components/lib/tasks/create-task.tsx]: Done!

**Review by aaronchongth — APPROVED** [review]
LGTM! Thanks!
